Enhance CI building for Mingw-w64 (Part 1)

SlySven opened the pull request:
This adds ~~three~~four shell scripts that try to replace some PowerShell ones to produce both 32 and 64-bit Windows packages. At present they should each produce a zip archive file containing all the files needed to execute the Mudlet application on a Windows PC that does NOT have MSYS2 installed in it. Further work is needed to get executable installers built from that set of files and to upload them to the *normal* location - at present they will be available as build artefacts on the AppVeyor CI platform...

Also: add some more existing files to QMake project file, so that they show up in the Qt Ctreator IDE!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: add some more existing files to QMake project file

So that they show up in Qt Ctreator IDE!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Enhance: setup CI building for Mingw-w64 (part 1)

This adds three shell scripts that try to replace some powershell ones to
produce both 32 and 64-bit Windows packages. At present they should each
produce a zip archive file containing all the files needed to execute the
Mudlet application on a Windows PC that does NOT have MSYS2 installed in
it. Further work is needed to get excecutable installers built from that
set of files and to upload them to the *normal* location - at present they
will be available as build artifacts on the AppVeyor CI platform...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: fix sh script errors and also ensure we set VERSION env. variable

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: remove section from script that is no longer needed

in the past I had used a trick in a script to get past an issue caused by
an overlarge font resources file that caused compilation failures. As that
file is no longer present the explict mention of it broke things!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 reviewed (commented):
Thanks a lot for this start! Could you rename the files so they keep the same naming structure as the macOS and Linux ones? So setup, build, after_success and such.

SlySven commented:
> Could you rename the files so they keep the same naming structure as the macOS and Linux ones? So setup, build, after_success and such.

I thought I had (apart, obviously from the file extension as they are shell scripts {NOT `bash` ones BTW}) - except for a *new* "after_build" one - which is not done in the same place in the CI process as the "after_success" stage ...?

vadi2 commented:
after_build does what after_success does now, right?

SlySven committed:
BugFix: throw build configuration switch needed for MSYS2/Mingw-w64 builds

I had forgotton it!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
Not completely - I need it currently as that is the point at which the *build artefacts* (a .zip file) can be captured and made available in the AppVeyor build web-pages. It may be that `after_success` could be used but that wasn't clear to me when I first came up with this set of scripts...

vadi2 commented:
Okay - well when it's ready, I'd like it to be consistent with other build system setup so the entire thing is easier to work on.

SlySven committed:
Revise: revise and refactor some AppVeyor sh scripts

So that the envirmental variables needed are available in all parts of the
CI build.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: work around a defect in windeployqt since Qt 5.14

Using the `--release` option breaks things!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
Oh blast, `windeployqt.exe` is borked since Qt 5.14 - the `--release` option doesn't work (it seems to be assumed unless `--debug` is given and it breaks if it ***is*** given): https://bugreports.qt.io/browse/QTBUG-84567

SlySven committed:
BugFix: reorder AppVeyor sh script to put "path" setup FIRST

Because of the wrong order the script was using a `tr` from a non-MSYS2
element that is also present on the CI platform (Cygwin) and there was
a dll mis-match.

Also included a missing project related environment variable in the build
environment that was needed.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: correct environmental variable declaration in yaml file

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: correct AppVeyer script to use AppVeyor environmental variables

I had copy/pasted a script from the Travis CI arrangement and forgotten to
tweak the CI variable names to suit!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: a typoo

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: correct test for being a PR build and tweak libraries list

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
Windows non-standard URLs:
Windows (32-bits): https://ci.appveyor.com/api/buildjobs/goxsx498nfvf2qx5/artifacts/package%2FMudlet-4.9.1-testing-pr4071-e0458c09-win32.zip
Windows (64-bits): https://ci.appveyor.com/api/buildjobs/fd3tj9bhecywtki8/artifacts/package%2FMudlet-4.9.1-testing-pr4071-e0458c09-win64.zip

SlySven commented:
Okay - main problem with those Windows builds is that they do not contain the Lua translation files that would be found in 
the `./translations/lua/translated/` folder (relative to the executable) of a Window's installation.

This might actually also be an issue for Linux packagers - as those files are NOT generated by the build process ~~and won't be present in the *source-code*~~ *Edit: they are, but are not listed in the qmake file so do not show up in the Qt Creator IDE*. For comparison the main `mudlet_xx_YY.qm` are generated from the corresponding `mudlet_xx_YY.ts` files during the build process (`qmake` or `cmake` based) but the latter are only there because we have a semi-automated process to get them from CrowdIn every so often - it looks as though ~~that process needs to be extended to include those Lua ones~~ *Edit: it seems like they are*!

Just added another commit to address this issue!

SlySven committed:
BugFix: add missing Lua translation files to build

Also add them to qmake project file so that they show up in Qt Creator!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: correct a path in previous commit

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented:
Also upload to make.mudlet.org/snapshots instead of appveyor, that's what we use :)

SlySven committed:
Revise: try a different method to copy Lua translation files in

I don't seem to be having as much sucess with rsync as I had in the past,
maybe because I was trying to be a bit more selective in my file selection.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
> Also upload to make.mudlet.org/snapshots instead of appveyor, that's what we use :)

I'm not uploading anything yet! The two zip files are currently being left there (for 6 months) as a "*build artefact*"... 😁

SlySven commented:
Nailed it! 🎉 
https://ci.appveyor.com/api/buildjobs/46sg70uy14n4p4a2/artifacts/package%2FMudlet-4.9.1-testing-pr4071-2cd102d5-win32.zip
https://ci.appveyor.com/api/buildjobs/a5x78u82mh5u8yjo/artifacts/package%2FMudlet-4.9.1-testing-pr4071-2cd102d5-win64.zip

vadi2 commented:
Is it squished a bit for you?

![Selection_256](https://user-images.githubusercontent.com/110988/93592521-9eaf3980-f9b2-11ea-88dc-d3ba2c76c714.png)

Leigheas commented:
Yeah I had noticed that too

SlySven commented:
Well, as I said on Discord, that looks to be a font issue - which is a nuisance but not totally unsurprising given the changes in the system parts that this PR will be introducing. It might be worth checked that we do not have *kerning* enabled whenever we are specifying a font - as that is not going to help when we are trying to be mono-spaced...

🔎 Also, notice that the previous release version (the 4.8.0 one) using the existing build system has the two left most main menu items overlapping each other - despite the remaining text items being better spaced!

From the Qt Documentation for `QFont`:
>### `void QFont::setKerning(bool enable)`
>Enables kerning for this font if `enable` is `true`; otherwise disables it. By default, kerning is enabled.
>When kerning is enabled, glyph metrics do not add up anymore, even for Latin text. In other words, the assumption that `width('a') + width('b')` is equal to `width("ab")` is not necessarily true.
>See also `kerning()` and `QFontMetrics`.

SlySven committed:
Revise: arrange for snap-shot builds to be uploaded to Mudlet server

Also remove unneeded space from empty sh script `echo` lines

Still need to do more complex building of (signed when we have a new
certificate in place!!!) "nuget"(?) packaged forms for
release/ptb type builds...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: restore Gitter notification and shorted SHA1 used in filename

Also ensure we actually call the `./CI/appveyor.after_success.sh` script!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: rename stage in appveyor.yml file to correct form

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: move to package directory BEFORE zipping things up

... otherwise we archive the whole project files, source code, git repo
and everything else. This was caused by move the archive file creation to
the new appveyor.after_success.sh script, it also means that it is no
longer possible to capture that archive and store it internally to AppVeyor
as a build artifact - so that bit needs to come out of the YAML control
file.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: fixup Gitter notification of deployment messages

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Testing: part of full Windows installer building done [skip travis]

This has been bodged for testing and is not fully operable so far - it
fakes a PTB build.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: correct typo in script

Also remove doubling of back-slashes in YAML file as it does not seem to be
necessary.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revert: the doubled back-slashes are needed after all [skip travis]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Enhance: try and add ccache to speed up (re)compilations

Not sure if this will work and given the 1GByte total cache availablity on
our AppVeyor platform (and using half of it for ccache) it may not be
enough, but if it DOES help for the recompilation of files that have not
changed since the last build (of a particular bitness) it could help a bit.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: add missing file change and fix == error in 'sh' test [skip travis]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: need to give full paths to `tr` and `perl` [skip travis]

Otherwise incompatible ones get used from Cygwin which is also present
in CI platform...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: more places where an explicit path is needed [skip travis]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Testing: check what path is being used [skip travis] [skip gha]

Also add a conditional to exclude GitHub Actions CI runs (as about or use
[skip ci] to stop all CI processes...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: correct a previous PATH setting error [skip travis] [skip gha]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: use better invocation of MSYS2 shells [skip travis] [skip gha]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revert: the commits to try and start MSYS2 shells [skip travis] [skip gha]

Trying to start them "properly" using the `msys2_shell.cmd` didn't work as
expected as the output was lost...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: one last try to get ccache working [skip travis] [skip gha]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revert: remove all attempt at using ccache [skip gha] [skip travis]

It just doesn't seem possible to use it...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

keneanung commented on .appveyor.yml:
```diff
@@ -4,10 +4,17 @@ branches:
+
+# Without a prefix commands are batch file ones
+# With cmd: prefix they are run by cmd.exe
+# With ps: prefix they are run by PowerShell
```
this can be found in the appveyor help, no need to document this here as well...

SlySven replied:
~~You have~~ ***I* had** to hunt to find it...!

vadi2 replied:
That's OK, you did find it! Doesn't mean we should copy it over

keneanung commented on .appveyor.yml:
```diff
@@ -19,22 +26,39 @@ environment:
   matrix:
-  - QT_BASE_DIR: C:\Qt\5.13.2\mingw73_32
-    MINGW_BASE_DIR: C:\Qt\Tools\mingw730_32
+# was QT_BASE_DIR: C:\Qt\5.13.2\mingw73_32 and MINGW_BASE_DIR: C:\Qt\Tools\mingw730_32 for old system and no BUILD_BITNESS
```
Please, no history here. If we need history, we use git.

SlySven replied:
This is still a WIP and I don't want to delete the prior stuff until I have glued *ALL* the new bits in.

keneanung commented on .appveyor.yml:
```diff
@@ -19,22 +26,39 @@ environment:
-  - QT_BASE_DIR: C:\Qt\5.13.2\mingw73_32
-    MINGW_BASE_DIR: C:\Qt\Tools\mingw730_32
+# was QT_BASE_DIR: C:\Qt\5.13.2\mingw73_32 and MINGW_BASE_DIR: C:\Qt\Tools\mingw730_32 for old system and no BUILD_BITNESS
+# We are not currently using CMake for Windows builds but it might be needed in
```
YAGNI: we will look into cmake when it's ready. Right now, leave the comment and Q_OR_C_MAKE out...

SlySven replied:
Who was it that was insisting that qmake was on the way out...? :thinking:

keneanung replied:
But we are not there yet. And once we go there, we replace `qmake` by `cmake` and don't need the variable either...

SlySven replied:
As it happens I'm now using it to pull the version number from the corresponding build system's project file.

keneanung commented on CI/appveyor.after_build.sh:
```diff
@@ -0,0 +1,187 @@
+
+echo "Copying system libraries in..."
+if [ "${BUILD_BITNESS}" = "32" ] ; then
+    cp -v -p ${MINGW_INTERNAL_BASE_DIR}/bin/libgcc_s_dw2-1.dll .
```
manually putting all dependencies here seems to be cumbersome... I am sure added a script to an issue, that automated the process using `ntldd`. I'd prefer that to listing all dependencies manually.

SlySven replied:
Trouble is I tried that - and found it doesn't pick up on the dynamically loaded libraries!

These at least includes: the Open SSL ones, the Discord RPC one and anything that is needed for the Lua subsystem... 

It would be nice to script something but I am not sure it will save us much effort. As it is, if something gets updated so that a file is NOT there will be an error in the build log - perhaps I need to include a status check on the copying commands so that the build job automatically fails if any of them do...

keneanung replied:
The dynamically linked libraries can be manually copied and ran with the same script to copies their dependencies over. That would gain us quite a bit, just look at your list.

> As it is, if something gets updated so that a file is NOT there will be an error in the build log - perhaps I need to include a status check on the copying commands so that the build job automatically fails if any of them do...

With the script, most of the time nothing needs to be changed around the CI scripts, which MUCH better in my opinion.

vadi2 replied:
Script for dependencies: https://github.com/Mudlet/Mudlet/pull/3255#issuecomment-681547576

SlySven committed:
Merge branch 'development' into Enhance_CiBuildingForMingw-w64_part1
To bring in recent changes in main development branch.

Signed-off by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Enhance_CiBuildingForMingw-w64_part1

SlySven committed:
Testing: a check to see that the AppVeyor CI can still build a snapshot

There are some revisions and some test behaviour but this is not complete
enough for anything other than PR snapshot builds. It aims to test to
see if messages can be sent to AppVeyor message listing and build artifacts
produced without listing them explicitly in the yaml file.

It also merges in recent commits in the development branch.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: a quoting type

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: another attempt to fix script errors [skip gha] [skip travis]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: correct CI build script error for AppVeyor [skip gha] [skip travis]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: more fixes for another CI Script [skip gha] [skip travis]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: remove explict requirement for python 2 [skip gha] [skip travis]

Python 2.x has finally left the MSYS2 / Mingw-w64 project - so we just need
to specify python to get python 3.x as asking for python2 no longer works.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: add Mingw-w64 package providing ntlld [skip gha] [skip travis]

`ntldd` was missing from package list but since I am now using it to
identify the shared libraries that Mudlet needs besides the Qt and some
other ones loaded dynamically.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

keneanung commented on CI/appveyor.after_build.sh:
```diff
@@ -0,0 +1,152 @@
+# continually trying to run the executable on the target type system
+# and adding in the libraries to the same directory and repeating that
+# until the executable actually starts to run. Alternatively running
+# ldd ./mudlet.exe | grep "/mingw32" {for the 32 bit case, use "64" for
```
This comment needs some tweaking now.

SlySven replied:
Agreed.

keneanung commented on CI/appveyor.after_build.sh:
```diff
@@ -0,0 +1,152 @@
+# process for individual librarys. For ones used by lua modules this
+# can manifest as being unable to "require" the library within lua
+# and doing the above "ldd" check revealed that "zip.dll" needed
+# "libzzip-0-13.dll" and "luasql/sqlite3.dll" needed "libsqlite3-0.dll"!
```
I think you will need to run the `ntldd` thing on the mentioned lua libraries as well.

SlySven replied:
Trouble was I didn't spot that I had to install the 32/64 bit package that provides it...

SlySven replied:
🤔 Checking the listed ones reveals the (handled) non-native lib requirements of:
* `lfs.dll`:
  * `lua51.dll`
* `lpeg.dll`
  * `lua51.dll`
* lsqlite3.dll
  * `lua51.dll`
  * `libsqlite3-0.dll`
* lua-utf8.dll
  * `lua51.dll`
* rex_pcre.dll
  * `lua51.dll`
  * `libpcre-1.dll`
* `yajl.dll`
  * `lua51.dll`
  * `libyajl.dll`
* `luasql/sqlite3.dll`
  * `lua51.dll`
  * `libsqlite3-0.dll`
* `brimworks/zip.dll` (changed to that one from the Kepler project's one in one of the last few commits)
  * `lua51.dll`
  * `libzip.dll`

I have run through the other libraries that are currently listed explicitly for various reasons and we seem to be covered. 🤞

keneanung replied:
> brimworks/zip.dll (changed to that one from the Kepler project's one in one of the last few commits)

Why? Keep to the scope, please :(

vadi2 replied:
Agree. This is scope leak.

SlySven replied:
:mag_right: :thinking: The other CIs are split over `lua-zip` v.s. `luazip`:
* MacOs - changed to it in https://github.com/Mudlet/installers/commit/fedaa2e8cd3476b573db2a46b2e75772eb03c9d8 18 Apr 2020 IIRC it was because the `luazip` version has a dependency on `zzliblip`/`libzzip`/`zzip` (the name seems to vary in different places but it does always have two "zz"s) that became problematic to compile on that platform at least.
* Linux  - refers to the library directly in the `.travis.yml` - confusingly the package is called `lua-zip` but it is the Kepler projects one (luarock `luazip`) as opposed to the Brimworks one (luarock `lua-zip`) - furthermore the *Debian* package concerned is based upon the version **1.2.3** code that was extracted from an SVN repository and place at https://github.com/luaforge/luazip where it has remained untouched since 2007. That code has been forked and revised at https://github.com/mpeterv/luazip - most recently in 2017 to a version **1.2.7** which is what the `luazip` rock is based upon. {So the `luazip` rock is based on code that is ten years newer than the Debian package!} in comparison the `lua-zip` rock is at a version **0.2.0** and is unrelated to the original Kepler project code - but it was last revised in 2019...

Given that I am constructing a new build system (and not compiling the luazip code from source as the prior Windows CI process did as it downloads and compiles https://github.com/rjpcomputing/luazip/archive/master.zip which is actually redirected to https://github.com/mpeterv/luazip {i.e. the source of the `luazip` rock NOT the older form that the Linux build uses}) I have to say, this ain't scope leak, but a reasonable choice out of a couple of alternatives **both of which our application is already built to handle**.

vadi2 replied:
It's changing more things than are needed - Windows was built with the other one so it would be safer to stick with it. But let's try it and see if it works well.

SlySven committed:
BugFix: switch to lua-zip (from luazip) rock [skip gha][skip travis]

We are preferring it over the other one as it does not depend on zziplib.

The other one missed getting into the previous commit for some reason.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented on CI/appveyor.install.sh:
```diff
@@ -22,7 +22,7 @@ if [ ${BUILD_BITNESS} != "32" ] && [ ${BUILD_BITNESS} != "64" ] ; then
 fi
 
-appveyor AddMessage "This is a \"${BUILD_TYPE}\" ${BUILD_BITNESS} bit Mudlet ${VERSION}${MUDLET_VERSION_BUILD} build ..."" -Catagory Information
+appveyor AddMessage "This is a \"${BUILD_TYPE}\" ${BUILD_BITNESS} bit Mudlet ${VERSION}${MUDLET_VERSION_BUILD} build ..." -Catagory Information
```
Category?

SlySven replied:
It is an argument to the "AddMessage" (displays a text on the AppVeyor build "Messages" tab) - with one of three values:
* Information
* Warning
* Error
see https://www.appveyor.com/docs/build-worker-api/#add-compilation-message ...

vadi2 replied:
Yeah but you spelled it as `Catagory`.

SlySven replied:
🤦‍♂️

SlySven committed:
Revise: partial revisions but NOT ready to go yet [skip gha][skip travis]

The snapshot build code should now be complete. The PT builds should also
work now. However the release build is still not complete. For testing
purposes the NEXT commit should try and use SCP to upload the snapshot
code so that it can be verified to be usable for the release builds - this
needs some extra encryption stuff and a key (and salt) will probably need
to be created to use the MSYS2 `scp` rather than the WINSCP NET module
that the original PowerShell implimentation used.

Includes a debug line to locate the secure-file utility that might be
needed.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Enhance_CiBuildingForMingw-w64_part1 [skip gha] [skip travis]

Conflicts resolved in:
* CI/travis.linux.after_success.sh
* CI/travis.osx.after_success.sh

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: use shellcheck to bugfix, check and reinforce shell scripts

I had used an `end` instead of `fi` to close an `if` and
https://www.shellcheck.net/ was helpful to find that AND strengthen all
the AppVeyor scripts I have produced.

IDEA: use it on the Linux and MacOs ones as well...! 8-)

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: repair luarock invocation (and improve Gitter notification)

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: take out unneeded uses of cygpath [skip gha][skip travis]

When using native (to MSYS2) paths (with '/' directory separators) it is
probably not necessary to use the $APPVAYOR_BUILD_DIR - we can hard code
the path '/c/projects/mudlet/` which makes the quoting and processing of
paths simpler - if harder to revise later one if that value ever changed.

Also needed to fix some uses of `rsync` with a wild-card, especially when
the path containing them is double-quoted. If rsync is processing the
wildcard (instead of the shell?) it will stop at directories if a single
`*` is used - apparently one needs to use s `**` to cause it to decend.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: correct some erroneous quoting and paths [skip gha] [skip travis]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: unify AppVeyor error messages & prepare fake PT build [skip gha] [skip travis]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: correct bash condition command execution [skip gha] [skip travis]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: correct script error caused by an extra '-' [skip gha] [skip travis]

Also swap 32 / 64 bit build job order to get results from latter first.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: correct typo ('x' in 64-bit filenames) [skip gha] [skip travis]

Also do not attempt to sign the Windows installer when we do not have the
password - which we won't have in a PR testing build like the one that
inserts this commit!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: forgot to double-quote a shell variable [skip gha] [skip travis]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: use scp to upload Win PTBuilds to server [skip gha][skip travis]

Parts of this can only be tested AFTER it has been merged in the
development branch because it uses some secure environment varables that
are NOT present for PR/testing builds.

HOWEVER ONCE IT HAS BEEN MERGED THINGS MUST BE REVERTED OR SUBQUENTLY
REVISED BEFORE THE development BRANCH IS USED FOR A RELEASE BUILD.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Enhance_CiBuildingForMingw-w64_part1

SlySven committed:
BugFix: fix some script errors [skip gha] [skip travis]

`pushd`/`popd` are bash shell built-ins and not present in a generic `sh`.

The string equality test in a generic shell `test` is `=` nor `==`.

One check for PT Builds used an old variable/logic which I had forgotten to
update.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: repair an error made in on-line merge [skip gha] [skip travis]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
`development` branch merged in and the code forked to a new branch: [`64Bit_Windows`](https://github.com/Mudlet/Mudlet/tree/64Bit_Windows).

SlySven commented:
Not sure now if this is the way to go - Vadim seems to have ~~cobbled together~~ produced a working GH Actions method to build a 64-Bit Windows Mudlet so that is probably a more desirable (on a speed basis if nothing else) basis to proceed. Given that I do prefer to work in a pure MSYS2/Mingw-w64 Windows environment and have no desires to touch the MSVC toolchain i will probably keep picking away at this at odd times...

SlySven committed:
Merge branch 'development' into Enhance_CiBuildingForMingw-w64_part1

Needed to pull in recent main line changes.

Conflicts resolved in:
* .appveyor.yml
* src/mudlet.pro

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Undo: remove fake PT build settings...

Forgot that I had left this in place... 8-(

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented:
Alright, sounds good. We don't need a PR for this then, it can stay in a branch.
